packages/medusa/src/subscribers/cart.ts

```diff
@@ -0,0 +1,70 @@
+import EventBusService from "../services/event-bus"
+import { CartService, PaymentProviderService } from "../services"
+import { EntityManager } from "typeorm"
+
+type InjectedDependencies = {
+  eventBusService: EventBusService
+  cartService: CartService
+  paymentProviderService: PaymentProviderService
+  manager: EntityManager
+}
+
+class CartSubscriber {
+  protected readonly manager_: EntityManager
+  protected readonly cartService_: CartService
+  protected readonly paymentProviderService_: PaymentProviderService
+  protected readonly eventBus_: EventBusService
+
+  constructor({
+    manager,
+    cartService,
+    paymentProviderService,
+    eventBusService,
+  }: InjectedDependencies) {
+    this.cartService_ = cartService
+    this.paymentProviderService_ = paymentProviderService
+    this.eventBus_ = eventBusService
+    this.manager_ = manager
+
+    this.eventBus_.subscribe(
+      CartService.Events.CUSTOMER_UPDATED,
+      async (cartId) => {
+        await this.onCustomerUpdated(cartId)
+      }
+    )
+  }
+
+  async onCustomerUpdated(cartId) {
+    await this.manager_.transaction(async (transactionManager) => {
```

Comment:
thought: Could this subscriber conflict with the call to create payment sessions i.e. the issue we currently have? 

Should it be serializable?

Reply:
Yes sure, is it retiable?